zhaquirks/custom/telink.py

```diff
@@ -0,0 +1,208 @@
+
+        # comfort temperature max: A value in 0.01ºC to set maximum comfort temperature for happy face
+        comfort_temperature_max = ZCLAttributeDef(
+            id=0x0103,
+            type=t.int16s,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # comfort humidity min: A value in 0.01%RH to set minimum comfort humidity for happy face
+        comfort_humidity_min = ZCLAttributeDef(
+            id=0x0104,
+            type=t.uint16_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # comfort humidity max: A value in 0.01%RH to set maximum comfort humidity for happy face
+        comfort_humidity_max = ZCLAttributeDef(
+            id=0x0105,
+            type=t.uint16_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # A value in 0.01ºC offset to fix up incorrect values from sensor
+        temperature_offset = ZCLAttributeDef(
+            id=0x0100,
+            type=t.int16s,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+        # A value in 0.01%RH offset to fix up incorrect values from sensor
+        humidity_offset = ZCLAttributeDef(
+            id=0x0101,
+            type=t.int16s,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+        # Measurement interval, step 1 second, range: 3..255 seconds. Default 10 seconds.
+        measurement_interval = ZCLAttributeDef(
+            id=0x0107,
+            type=t.uint8_t,
+            access="rw",
+            is_manufacturer_specific=True,
+        )
+
+
+(
+    QuirkBuilder("MiaoMiaoCe", "MHO-C401N-z")
+    .applies_to("MiaMiaoCe", "MHO-C401N-z")  # typo until v. 1.2.2
+    .applies_to("MiaoMiaoCe", "MHO-C401-z")
+    .applies_to("MiaoMiaoCe", "MHO-C122-z")
+    .applies_to("MiaMiaoCe", "MHO-C122-z")  # typo until v. 1.2.2
+    .applies_to("Xiaomi", "LYWSD03MMC-z")
+    .applies_to("Tuya", "TS0201-z")
+    .applies_to("Tuya", "TH03Z-z")
+    .applies_to("Tuya", "ZTH01-z")
+    .applies_to("Tuya", "ZTH02-z")
+    .applies_to("Tuya", "LKTMZL02-z")
+    .applies_to("Sonoff", "TH03-z")
+    .applies_to("Qingping", "CGDK2-z")
+    .removes(CustomUserInterfaceCluster.cluster_id, cluster_type=ClusterType.Client)
+    .adds(CustomUserInterfaceCluster)
+    .number(
+        CustomUserInterfaceCluster.AttributeDefs.temperature_offset.name,
+        CustomUserInterfaceCluster.cluster_id,
+        min_value=-327.67,
+        max_value=327.67,
+        step=0.01,
+        unit=UnitOfTemperature.CELSIUS,
+        translation_key="temperature_offset",
+        fallback_name="Temperature offset",
+        multiplier=0.01,
+        mode="box",
+    )
+    .number(
+        CustomUserInterfaceCluster.AttributeDefs.humidity_offset.name,
+        CustomUserInterfaceCluster.cluster_id,
+        min_value=-327.67,
+        max_value=327.67,
+        step=0.01,
+        # unit=PERCENTAGE,
```

Comment:
Support for all units was merged in ZHA with https://github.com/zigpy/zha/pull/352, so we can add this now (and below), as the next HA/ZHA versions will support this.